src/main/java/net/starschema/clouddb/jdbc/BQSupportFuncts.java

```diff
@@ -158,69 +158,29 @@ public static Properties getUrlQueryComponents(String url, Properties defaults)
-        projectId = projectId.replace("__", ":").replace("_", ".");
-        JobReference completedJobReference = completedJob.getJobReference();
-        GetQueryResultsResponse queryResult = bigquery
-                .jobs()
-                .getQueryResults(projectId, completedJobReference.getJobId())
-                .setLocation(completedJobReference.getLocation())
-                .execute();
-        List<TableRow> rows = queryResult.getRows();
-        System.out.print("\nQuery Results:\n------------\n");
-        for (TableRow row : rows) {
-            for (TableCell field : row.getF()) {
-                System.out.printf("%-20s", field.getV());
-            }
-            System.out.println();
-        }
-    }
-
     /**
      * Return a list of Projects which contains the String catalogname
      *
-     * @param catalogName The String which the id of the result Projects must contain
+     * @param projectId The String which the id of the result Projects must contain
      * @param Connection  A valid BQConnection instance
      * @return a list of Projects which contains the String catalogname
      * @throws IOException <p>
      *                     if the initialization of requesting the list of all Projects
      *                     (to be sorted from) fails
      *                     </p>
      */
-    private static List<Projects> getCatalogs(String catalogName,
+    private static List<Projects> getProjects(String projectId,
                                               BQConnection Connection) throws IOException {
-        logger.debug("Function call getCatalogs catalogName: " +
-                (catalogName != null ? catalogName : "null"));
+        logger.debug("Function call getProjects catalogName: " +
+                (projectId != null ? projectId : "null"));
         List<Projects> projects = Connection.getBigquery().projects().list()
                 .execute().getProjects();
 
-        //since we'll use . -> _ : -> __ conversion, it's easier to
-        // replace all the . : before we do any compare
         if (projects != null && projects.size() != 0) {             //we got projects!
-            for (Projects projects2 : projects) {
-                projects2.setId(projects2.getId().replace(".", "_").replace(":", "__"));
-                //updating the reference too
-                ProjectReference projRef = projects2.getProjectReference();
-                projRef.setProjectId(projRef.getProjectId().replace(".", "_").replace(":", "__"));
-                projects2.setProjectReference(projRef);
-            }
-            if (catalogName != null) {
+            if (projectId != null) {
                 List<Projects> ProjectsSearch = new ArrayList<Projects>();
                 for (Projects project : projects) {
-                    if (project.getId().contains(catalogName)) {
+                    if (project.getId().equals(projectId)) {
```

Comment:
@kalenp I think this method can be rethought.
How can this ever return a List now -- there should only be **1**.

It should either simply return the BigQuery _Project_ model or itself be a no-op now.

Reply:
If `projectId == null`, then this will return the full list of projects.  See https://github.com/jonathanswenson/bqjdbc/pull/71/files/48675ef14ed622ebf9267d694f6d3445c2e4d875#diff-72ff40b7727288cb223d0640acec93efc2e28f70bd903b0b758a99f44a5f5511L233 below.

Reply:
That commit link doesn't work anymore with the force push but the API here is pretty nonsensical